### PR TITLE
Moved to ginkgo based tests in scaffolding 2.0

### PR DESCRIPTION
**Pull request by droot**
With scaffolding version 2.0, 
 - scaffolded tests for API types use ginkgo.
 - Re-enabled scaffolding for simple CRD manifest until we have crd-generation working for scaffolding 2.0
 - Added one more kind in test project to enable testing for multiple kinds.

### CODE REVIEW

File: test/project_v2/main.go [hunk 1]
```diff
@@ -37,6 +37,7 @@ var (
 
 func init() {
 
+	crewv1.AddToScheme(scheme)
 	crewv1.AddToScheme(scheme)
```
Comment (droot) [inline test/project_v2/main.go]: This is a bug where `crewv1.AddToScheme` is getting duplicated. Will fix this in a separate PR.

**Comment by droot** [comment]
@DirectXMan12 Can you pl. take a look at this one.

File: generated_golden.sh [hunk 2]
```diff
@@ -52,6 +52,7 @@ scaffold_test_project() {
 		$kb create api --group policy --version v1beta1 --kind HealthCheckPolicy --example=false --controller=true --resource=true --namespaced=false --make=false
 	elif [ $version == "2" ]; then
 		$kb create api --group crew --version v1 --kind FirstMate --controller=true --resource=true --make=false
+		$kb create api --group crew --version v1 --kind SecondMate --controller=true --resource=true --make=false
```
Comment (mengqiy) [inline generated_golden.sh]: I use `crew Captain` in https://github.com/kubernetes-sigs/kubebuilder/pull/644/files#diff-59d90b4ae2a1649c75c68b352dcb3a49R54.

File: test/project_v2/api/v1/firstmate_types_test.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+	"golang.org/x/net/context"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+//
+// These tests are written in BDD-style using Ginkgo framework. Refer to
+// http://onsi.github.io/ginkgo to learn more.
+//
```
Comment (mengqiy) [inline test/project_v2/api/v1/firstmate_types_test.go]: nit: these 2 extra `//` make it looks funny. I'd suggest using either
```
// blah
// blah
```
or
```
/////////////
// blah    //
// blah    //
/////////////
```

File: test/project_v2/api/v1/firstmate_types_test.go [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+	AfterEach(func() {
+		// Add any teardown steps that needs to be executed after each test
+	})
+
+	//
+	// Add Tests for OpenAPI validation (or additonal CRD features) specified in
+	// your API definition.
+	// Avoid adding tests for vanilla CRUD operations because they would
+	// test Kubernetes API server, which isn't the goal here.
+	//
```
Comment (mengqiy) [inline test/project_v2/api/v1/firstmate_types_test.go]: Same here

File: test/project_v2/api/v1/secondmate_types_test.go [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+	"golang.org/x/net/context"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+//
+// These tests are written in BDD-style using Ginkgo framework. Refer to
+// http://onsi.github.io/ginkgo to learn more.
+//
```
Comment (mengqiy) [inline test/project_v2/api/v1/secondmate_types_test.go]: Same here

File: test/project_v2/api/v1/secondmate_types_test.go [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+	AfterEach(func() {
+		// Add any teardown steps that needs to be executed after each test
+	})
+
+	//
+	// Add Tests for OpenAPI validation (or additonal CRD features) specified in
+	// your API definition.
+	// Avoid adding tests for vanilla CRUD operations because they would
+	// test Kubernetes API server, which isn't the goal here.
+	//
```
Comment (mengqiy) [inline test/project_v2/api/v1/secondmate_types_test.go]: Same here

File: test/project_v2/api/v1/suite_test.go [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+)
+
+//
+// These tests use Ginkgo (BDD-style Go testing framework). Refer to
+// http://onsi.github.io/ginkgo/ to learn more about Ginkgo.
+//
```
Comment (mengqiy) [inline test/project_v2/api/v1/suite_test.go]: Same here

File: generated_golden.sh (hunk 2 above)
Comment (mengqiy) [reply generated_golden.sh]: We can drop this line now.
You will see `crew Captain` after rebase.

Comment (droot) [reply generated_golden.sh]: Removed the `SecondMate` kind and rebased.

File: test/project_v2/api/v1/firstmate_types_test.go (hunk 3 above)
Comment (droot) [reply test/project_v2/api/v1/firstmate_types_test.go]: killed the lines with empty comment.

File: test/project_v2/api/v1/firstmate_types_test.go (hunk 4 above)
Comment (droot) [reply test/project_v2/api/v1/firstmate_types_test.go]: Fixed all of these.

File: test/project_v2/api/v1/suite_test.go [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+func TestAPIs(t *testing.T) {
+	RegisterFailHandler(Fail)
+
+	RunSpecs(t, "v1 Suite")
+}
+
+var _ = BeforeSuite(func() {
+	By("bootstrapping test environment")
+	testEnv = &envtest.Environment{
+		CRDDirectoryPaths: []string{filepath.Join("..", "..", "config", "crds")},
```
[the remark this answers is not in the document]
Comment (droot) [reply test/project_v2/api/v1/suite_test.go]: Ack.

**Pull request merged**